### ISSUE TITLE
Compatibility with Rails 5.0

### ISSUE THREAD

**Issue by mvastola** [rails/rails-observers#37]
With the release of Rails 5.0.0.beta1 just days ago, many users (myself included) are seeking to upgrade their current Rails 4.2 apps to ensure forward-compatibility with this new version of Rails and to make use of its new features and capabilities.

rails-observers version 0.1.3.alpha is forward-compatible with Rails 5, but it has not been pushed to RubyGems, largely preventing usage of this gem (or even experimentation with it) in this new version of Rails as the latest released version of rails-observers (0.1.2) is locked to the Rails 4.x series.

I would like to request that a new version of rails-observers be pushed to RubyGems to mark the beginning of Rails 5 compatibility. (Ideally as a release, rather than an alpha version, but anything would be an improvement.)

**Comment by rafaelfranca** [comment]
That is the beauty of bundler. You can point to a git repository and don't wait for a release in Rubygems.

**Comment by mvastola** [comment]
Also:
* When doing some investigating prior to making this request, I wanted to see if rails-observers' test suite passed for Rails 5.0 (as this would be evidence to justify a non-alpha release), but I abandoned this pursuit when I found I actually can't get the test suite to even pass with Rails 4.2.5. Is this currently broken?
* Please note that rails-observers is a runtime dependency of the popular [audited](https://rubygems.org/gems/audited) gem, and is [blocking](https://github.com/collectiveidea/audited/issues/249) that gem from releasing a Rails 5 compatible version.

**Comment by mvastola** [comment]
@rafaelfranca I'm aware of the many wonders of bundler. :-)
However, due to the aforementioned testing issues, there's no reliable way to confirm the gem is fully-compatible with Rails 5, and, although there's nothing prohibiting it, it's probably a bad idea for a gem to depend on a commit to a github repo (nevermind a potentially unstable alpha release). Given that, I'm not really comfortable asking (in good faith) a gem developer to adopt such a bad practice even if they were willing.

**Comment by rafaelfranca** [comment]
Feel free to open a PR adding compatibility with Rails 5.

**Comment by mvastola** [comment]
Thanks for reopening. I actually can't tell if additional commits are needed to make the gem Rails 5 compatible because right now the test suite is failing for me even with Rails 4.2. Can you tell me if this is a known problem and/or if there is a known Gemfile.lock with which the tests pass? 

I tried making some changes to the tests (https://github.com/Partyista/rails-observers) and got rid of a bunch of warnings and deprecations so far, but I haven't figured out the rest yet.

I think only once the tests are working on 4.2 is it feasible to judge if Rails 5 is working by running the tests on that platform.

**Comment by rafaelfranca** [comment]
I can't tell if the gem is a good state too. There is no active development since July. I recommend you to give it a try and open pull request. Otherwise you would have to wait someone have time and interest in working on this gem, what can take some time.

**Comment by mvastola** [comment]
Ok. I'll PR what I have so far then. I don't mind trying to put some time into the gem myself to get the tests passing.. was just hoping you might have guidance (and didn't want to go this route if it were just an issue on my end).

**Comment by mvastola** [comment]
FYI, I'm temporarily closing my pull request because I think I can fix everything. I figured out that the tests pass on Rails 4.0.x, but failures start cropping up in 4.1.x, so I'm going to try to trace them and fix them before submitting a new PR.

**Comment by mvastola** [comment]
@rafaelfranca Please see PR #39, which completely (as far as I can tell) resolves all testing issues for the latest release for each of the following minor versions of Rails: 4.0, 4.1, 4.2, 5.0 (beta release).

I therefore request that it be merged and the version of rails-observers be bumped and it be pushed to rubygems.

**Comment by mvastola** [comment]
@rafaelfranca I have further updated #39 (which is once again ready to go) to include the `appraisal` gem to have Travis run its CI tests against a variety of Ruby/Rails version tuples. If the addition of this gem is undesirable, let me know, but I found it useful to catch a couple more of minor issues in the tests.

Everything is once again passing and ready to merge and I don't expect needing to change the PR again (barring any requests from yourself).

Thanks a ton,
Mike

**Comment by Wolfe13** [comment]
+1

**Comment by markaschneider** [comment]
+1

**Comment by kannans** [comment]
@rafaelfranca   👍
